test/promql/tests/application_errors_test.yaml

```diff
@@ -0,0 +1,102 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.application_alerts.yaml
+
+tests:
+  # Alerted cases:
+  - interval: 5m
+    input_series:
+
+      # HAS is experiencing application deletion failure rate of 100%, so it will be alerted
+      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1 2 3'
+      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1 2 3'
+
+    alert_rule_test:
+      - eval_time: 65m
+        alertname: ApplicationDeletionErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: application-service
+              pod: has
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                HAS is experiencing application deletion failures of >1%
+              description: >-
+                Application controller in Pod has for namespace
+                application-service on cluster cluster01 is failing to
+                successfully delete at least 99% of applications over the past hour
+              runbook_url: TBD
+
+  - interval: 5m
+    input_series:
+
+      # HAS experienced 10% deletion failure rate, should still alert
```

Comment:
nitpicking: this is around ~17% at the time of evaluation time. If you evaluate it at 100m, it will actually be 0%.
Can either change the comment, or the time series. Same goes for the 5% test case.

Reply:
So I'm trying to get a simple test case that will match ~10% failure rate, but I'm not sure why this isn't alerting?
```
      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
        values: '10+10x64'
      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
        values: '1+1x64'
```